packages/clay-color-picker/src/GradientSelector.tsx

```diff
@@ -0,0 +1,78 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+import React, {useEffect, useRef} from 'react';
+import tinycolor from 'tinycolor2';
+
+import {colorToXY, xToSaturation, yToVisibility} from './util';
+import {useMousePosition} from './hooks';
+
+interface GradientSelectorProps {
+	color: tinycolor.Instance;
+	hue?: number;
+	onChange?: (saturation: number, visibility: number) => void;
+}
+
+/**
+ * Renders GradientSelector component
+ */
+const GradientSelector: React.FunctionComponent<GradientSelectorProps> = ({
```

Comment:
I think we have some problems with using mouse position when I try to move to the lower right corner.

<img src="http://g.recordit.co/qa21UQk3vb.gif" />

Reply:
Yeah the problem is that both corners are the same hex value. Let me see if I can figure something out with this, its a bit tricky,

Reply:
It seems a lot better now, there's a case that might not be a problem. I can now move to the lower right corner, but when I move in the range hue, the Pointer goes back to the other side, this will happen with any update on the component.

Reply:
I think that issue might be unavoidable without writing specific code to handle `#000000` differently than others. What do you think of leaving that in now, since it doesn't effect the actual value, and then addressing it later if it becomes more of an issue?

Reply:
Yeah, no problem, it does not affect user usability.